media-source/mediasource-activesourcebuffers.html

```diff
@@ -0,0 +1,212 @@
+                  });
+              });
+          }, "SourceBuffer added to activeSourceBuffers list when its only audio track gets loaded (and thus becomes enabled).");
+
+
+          mediasource_test(function(test, mediaElement, mediaSource)
+          {
+              mediaElement.addEventListener("error", test.unreached_func("Unexpected event 'error'"));
+              MediaSourceUtil.fetchManifestAndData(test, manifestFilenameVideo, function (typeVideo, dataVideo)
+              {
+                  var sourceBuffer = mediaSource.addSourceBuffer(typeVideo);
+                  assert_equals(mediaSource.activeSourceBuffers.length, 0,
+                    "activeSourceBuffers is empty to start with");
+
+                  test.expectEvent(mediaElement, "loadedmetadata");
+                  test.expectEvent(mediaSource.activeSourceBuffers, "addsourcebuffer");
+                  sourceBuffer.appendBuffer(dataVideo);
+
+                  test.waitForExpectedEvents(function()
+                  {
+                      assert_equals(mediaSource.activeSourceBuffers.length, 1,
+                        "activeSourceBuffers updated when media element is loaded");
+                      assert_equals(mediaSource.activeSourceBuffers[0], sourceBuffer,
+                        "activeSourceBuffers contains sourceBuffer when media element is loaded");
+                      test.done();
+                  });
+              });
+          }, "SourceBuffer added to activeSourceBuffers list when its only video track gets loaded (and thus becomes selected).");
+
+
+          mediasource_test(function(test, mediaElement, mediaSource)
+          {
+              mediaElement.addEventListener("error", test.unreached_func("Unexpected event 'error'"));
+              MediaSourceUtil.fetchManifestAndData(test, manifestFilenameAudio, function (typeAudio, dataAudio)
+              {
+                  MediaSourceUtil.fetchManifestAndData(test, manifestFilenameVideo, function (typeVideo, dataVideo)
+                  {
+                      var sourceBufferAudio = mediaSource.addSourceBuffer(typeAudio);
+                      var sourceBufferVideo = mediaSource.addSourceBuffer(typeVideo);
+                      assert_equals(mediaSource.activeSourceBuffers.length, 0,
+                        "activeSourceBuffers is empty to start with");
+                      assert_equals(mediaSource.sourceBuffers.length, 2,
+                        "sourceBuffers list contains both SourceBuffers");
+                      assert_equals(mediaSource.sourceBuffers[0], sourceBufferAudio,
+                        "first SourceBuffer is the audio one");
+                      assert_equals(mediaSource.sourceBuffers[1], sourceBufferVideo,
+                        "second SourceBuffer is the video one");
+
+                      test.expectEvent(mediaElement, "loadedmetadata");
+                      test.expectEvent(mediaSource.activeSourceBuffers, "addsourcebuffer");
```

Comment:
There should be two 'addsourcebuffer' events fired (one for each of the active sourceBuffers).